src/services/types.ts

```diff
@@ -712,19 +712,16 @@ namespace ts {
         entries: CompletionEntry[];
     }
 
+    // see comments in protocol.ts
     export interface CompletionEntry {
         name: string;
         kind: ScriptElementKind;
-        kindModifiers: string;   // see ScriptElementKindModifier, comma separated
+        kindModifiers: string;
         sortText: string;
-        /**
-         * An optional span that indicates the text to be replaced by this completion item. It will be
-         * set if the required span differs from the one generated by the default replacement behavior and should
-         * be used in that case
-         */
         replacementSpan?: TextSpan;
         hasAction?: true;
         source?: string;
+        isRecommended?: true;

```

Comment:
i would rather we put this on the CompletionInfo instead. i understand for the server we have to put it on the entry.. but for VS for example, it makes more sense to have it on the info..
Please check with @amcasey about this too.

Reply:
I'm not intimately familiar with these types.  The info is the wrapper object containing a list of entries?  Assuming that's the case, having it on the entry is the most convenient place for VS.

Reply:
It's defined at `interface CompletionInfo` in `services/types.ts`

Reply:
Then, yes, the entry would be preferable to the info.  I've created a PR with the VS change if you'd like to test it.